File: 784-InsertIntoABinarySearchTree/784-InsertIntoABinarySearchTree.py
# Last updated: 2/9/2026, 9:54:45 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        
        tracker = root
        
        while tracker:
            
            if tracker.val < val and tracker.right is None:
                tracker.right = TreeNode(val)
                return root
            
            elif tracker.val > val and tracker.left is None:
                tracker.left = TreeNode(val)
                return root 
            
            elif tracker.val < val:
                tracker = tracker.right
            
            else:
                tracker = tracker.left
        
        root = TreeNode(val)
        return root 

File: 784-InsertIntoABinarySearchTree/784-InsertIntoABinarySearchTree.java
// Last updated: 2/9/2026, 9:54:48 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        if(root == null) return new TreeNode(val);
        TreeNode current_val = root;
        while(true){
            if(current_val.val > val){
                if(current_val.left != null){
                    current_val = current_val.left;
                }
                else{
                    current_val.left = new TreeNode(val);
                    break;
                }
            }
            if(current_val.val < val){
                if(current_val.right != null){
                    current_val = current_val.right;
                }
                else{
                    current_val.right = new TreeNode(val);
                    break;
                }
            }
        }
        return root;
    }
}